src/helpers/db.js

```diff
@@ -0,0 +1,136 @@
+import sqlite3 from 'sqlite3';
+
+const {
+    DB_FILE = 'data/database.db',
+} = process.env;
+
+// Open db
+let db = new sqlite3.Database(DB_FILE);
+
+/**
+ * Function to create required DBs if they do not exist
+ */
+export const createDBs = async () =>
+{
+    await db.exec("PRAGMA foreign_keys=ON");
+    await db.exec(`
+        CREATE TABLE IF NOT EXISTS jobs (
+        id TEXT,
+        name TEXT PRIMARY KEY,
+        request_id INTEGER DEFAULT 0,
+        last_reported_round INTEGER DEFAULT 0,
+        last_request_sent REAL DEFAULT 0,
+        last_submission_time REAL DEFAULT 0,
+        last_result REAL DEFAULT -1,
+        last_received_request_id INTEGER DEFAULT 0
+        );
+
+        CREATE TABLE IF NOT EXISTS rounds (
+        feed TEXT,
+        round_id INTEGER DEFAULT 0,
+        started_at REAL DEFAULT 0,
+        started_by TEXT,
+        submission_made INTEGER DEFAULT 0,
+        FOREIGN KEY (feed) REFERENCES jobs(name) ON DELETE CASCADE
+        );
+  `);
+}
+
+/**
+ * Function to get all jobs from the DB
+ * @returns array of jobs in DB
+ */
+export const getAllJobs = async () => {
+    return new Promise((resolve, reject) => {
+        db.all("SELECT * FROM jobs", (err, rows) => {
+            if (err) {
+            console.log("DB ERROR:", err)
+            reject([]);
+            } else {
+            resolve(rows);
+            }
+        });
+    });
+}
+
+/**
+ * Function to add a job to the DB
+ * @param {*} id id of the job
+ * @param {*} name name of the feed
+ */
+export const createJob = async (id, name) => {
+    try {
+        await db.run('INSERT INTO jobs (id, name) VALUES (?, ?)', [id, name])
+        await db.run('INSERT INTO rounds (feed) VALUES (?)', [name])
```

Comment:
Ye this is SQL injection hell but I think it's fine since the node is internally and only the node operator has access?

Reply:
SQL injection is only an issue when having a server and a DB. In this scenario, each oracle will have their own database and I doubt they would try and break their own DB.